File: flutter_app/lib/data/prompt_privacy.dart
const promptMetadataKeys = {
  'hasMission',
  'missionCompleted',
  'kissedPrivate',
  'whoKissedWhoPrivate',
};

String privacyKeyFor(String promptId) => '${promptId}Private';

bool isPromptMetadataKey(String key) {
  return promptMetadataKeys.contains(key) || key.endsWith('Private');
}

bool isPromptPrivate(Map<String, dynamic> prompts, String promptId) {
  if (promptId == 'kissedAnyone' || promptId == 'kissedWho') {
    return prompts['kissedPrivate'] == true ||
        prompts['whoKissedWhoPrivate'] == true;
  }
  return prompts[privacyKeyFor(promptId)] == true;
}

bool kissedAnyoneValue(dynamic value) {
  if (value == true) return true;
  if (value is String) return value.toLowerCase() == 'yes';
  return false;
}

bool hasMissionValue(dynamic value) => value == true;

File: src/lib/prompt-privacy.ts
const METADATA_KEYS = new Set([
  "hasMission",
  "missionCompleted",
  "kissedPrivate",
  "whoKissedWhoPrivate",
]);

export function privacyKeyFor(promptId: string): string {
  return `${promptId}Private`;
}

export function isPromptMetadataKey(key: string): boolean {
  return METADATA_KEYS.has(key) || key.endsWith("Private");
}

export function isPromptPrivate(
  prompts: Record<string, unknown> | undefined,
  promptId: string
): boolean {
  if (!prompts) return false;
  if (promptId === "kissedAnyone" || promptId === "kissedWho" || promptId === "whoKissedWho") {
    return prompts.kissedPrivate === true || prompts.whoKissedWhoPrivate === true;
  }
  return prompts[privacyKeyFor(promptId)] === true;
}
